portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +26,27 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  ArrayList<String> messages = new ArrayList<String>();
```

Comment:
```suggestion
  private List<String> messages = new ArrayList<String>();
```

Two common best practices here:
1. keep variables and methods visibility as low as possible. If this variable is only used by methods in this class, make it private so that other classes won't be able to access it by mistake https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html. 

2. When declaring the variable or defining methods, we usually assign the type to a more general interface, for example, List in this case. This makes it compatible with more methods. Imagine there is a helper method that takes in LinkedList, this ArrayList variable can not be passed into this methods unless converting ArrayList to LinkedList. But if both variable and the helper method takes List, it can be used without converting.

Reply:
Thank you for pointing this out to me, while I was debugging I was checking to see if the accessibility was causing errors so I must have forgot to change it back to private. Just fixed this!